cmake/algebra-plugins-compiler-options.cmake

```diff
@@ -4,6 +4,15 @@
 #
 # Mozilla Public License Version 2.0
 
+# Include the helper function(s).
+include( algebra-plugins-functions )
+
 # Set up the used C++ standard.
 set( CMAKE_CXX_STANDARD 17 CACHE STRING "The (host) C++ standard to use" )
 set( CMAKE_CUDA_STANDARD 17 CACHE STRING "The (CUDA) C++ standard to use" )
+
+# Turn on the correct setting for the __cplusplus macro with MSVC.
+if( "${CMAKE_CXX_COMPILER_ID}" MATCHES "MSVC" )
+   algebra_add_flag( CMAKE_CXX_FLAGS "/Zc:__cplusplus" )
+   algebra_add_flag( CMAKE_CUDA_FLAGS "-Xcompiler /Zc:__cplusplus" )
+endif()
```

Comment:
A note to @stephenswat and @paulgessinger: I guess we should rather set up this flag (`/Zc:__cplusplus`) as a public compile flag on `vecmem::core`. :thinking: Since MSVC will always need to use this flag when looking at the headers from that library.

I'll play with this a bit, but something like

```cmake
target_compile_options( vecmem_core PUBLIC
   $<$<AND:$<CXX_COMPILER_ID:MSVC>,$<COMPILE_LANGUAGE:CXX>>, /Zc:__cplusplus>
   $<$<AND:$<CXX_COMPILER_ID:MSVC>,$<COMPILE_LANGUAGE:CUDA>>, -Xcompiler /Zc:__cplusplus> )
```

should work... :thinking: